src/Admin/SiteHealth.php

```diff
@@ -171,6 +207,111 @@ public function persistent_object_cache() {
 		];
 	}
 
+	/**
+	 * Return list of available object cache mechanism.
+	 *
+	 * @return array
+	 */
+	public function check_available_cache() {
+		return [
+			'redis'     => $this->is_site_has_redis(),
+			'memcached' => $this->is_site_has_memcached(),
+			'apcu'      => $this->is_site_has_apcu(),
+		];
+	}
+
+	/**
+	 * Check if site has redis cache mechanism.
+	 *
+	 * @return bool True if redis cache is available otherwise False.
+	 */
+	public function is_site_has_redis() {
+
+		if ( ! class_exists( 'Redis' ) ) {
+			return false;
+		}
+
+		/*
+		 * Try to connect redis server.
+		 */
+		$redis_server = [
```

Comment:
I see that W3 Total Cache only [checks if the `Redis` class exists](https://github.com/W3EDGE/w3-total-cache/blob/d4c499acee517dc951dda093cc6cb99c547ff130/Util_Installed.php#L98-L100).

Any reason to go further and attempt to ping the Redis server? It seems like a brittle way of determining if there is indeed a Redis server as there may be many things that could cause it not to respond such as a firewall or iptables rule, for instance. It may also come off as hostile to some persons as it's attempting to make a connection that the user did not deliberately agree to.

Reply:
@pierlon As I understood, We do need to check services with connecting instead of checking if a class of extension exists. And if some services are not able to connect then it's better not to recommend anything.

Reply:
Right, but attempting to connect and fail would do more harm than good here and is more likely to produce more false positives. We should be only checking if a user's environment is capable of using Redis cache, not if we can connect to a Redis server.